### ISSUE TITLE
Ubuntu compile with error

### ISSUE THREAD

**Issue by vit1251** [mailru/icqdesktop#9]
Hello,

I try to compile ICQ on Ubuntu platform and have follow problem:

```bash
g++ -m64 -pipe -std=c++0x -O2 -Wall -W -D_REENTRANT -DSTRIP_VOIP -DINTERNAL_RESOURCES -DQT_NO_DEBUG -DQT_MULTIMEDIA_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -DQT_SHARED -I/usr/share/qt4/mkspecs/linux-g++-64 -I. -I/usr/include/qt4/QtCore -I/usr/include/qt4/QtNetwork -I/usr/include/qt4/QtGui -I/usr/include/qt4/QtMultimedia -I/usr/include/qt4 -I. -I../external/OpenAl/include -I../external/ffmpeg/include -I. -x c++-header -c stdafx.h -o icq.gch/c++
stdafx.h:1:9: warning: #pragma once in main file
 #pragma once
         ^
stdafx.h:166:27: fatal error: QtGui/qscreen.h: No such file or directory
 #include <QtGui/qscreen.h>
                           ^
compilation terminated.
Makefile:662: recipe for target 'icq.gch/c++' failed
make: *** [icq.gch/c++] Error 1
```

I make shell 

```
$ qmake gui.pro
$ make
```

What's wrong with this situation? Do u have a installation docs?

**Comment by mikhailnov** [comment]
Read README.md more carefully, Qt has to be patched (what for?!) and built.

**Comment by vit1251** [comment]
@mikhailnov I does not found workflow to compile in README.md just only suggest! If patching is really require, then why does not ask Qt Team about include this patch? I does not want to patch my system libqt4-dev and I am sure it totally does not require.

**Comment by rzhilkin** [comment]
Qt patch doesn't affect the build. 
We build it via Qt creator, which provides a path to Qt includes by itself. So it looks like you need to add a path to Qt includes to INCLUDEPATH variable in gui.pro file. However, we are going to upload Qt binary to github soon (includes and libs), like in Mac version. So you can just waiting for it.

**Comment by maksbotan** [comment]
In fact, this problem is caused by using Qt4's qmake. Run with Qt5 qmake and collect more awesome errors :)

**Comment by vit1251** [comment]
Let's try to provide script's to install on Ubuntu:

```bash
# apt-get install qt5-qmake
# apt-get install libqt5multimedia5
```

**Comment by vit1251** [comment]
New error with follow problem

```
utils/utils.cpp: In function ‘const QString& Utils::appFontFamily(Utils::FontsFamily)’:
utils/utils.cpp:1018:19: error: ‘class QSysInfo’ has no member named ‘windowsVersion’
    if (QSysInfo().windowsVersion() >= QSysInfo::WV_VISTA)
                   ^
utils/utils.cpp:1018:39: error: ‘WV_VISTA’ is not a member of ‘QSysInfo’
    if (QSysInfo().windowsVersion() >= QSysInfo::WV_VISTA)
```

**Comment by rzhilkin** [comment]
I dunno which version of Qt introduced this method declaration for non windows platforms. I see it in 5.5.1, while in 5.4 it is not available. We are using 5.5.1 version to compile, so you need to switch Qt version or avoid this error by ifdef construction(__linux__). For linux, function should place "Arial" in result map.

**Comment by rzhilkin** [comment]
> Let's try to provide script's to install on Ubuntu:

Let's try to read README.md and not to reinvent the wheel.

**Comment by vit1251** [comment]
> Let's try to read README.md and not to reinvent the wheel.

Unfortunately, the documentation is very sparse and clearly do not understand and do not specify version dependencies, and therefore have to investigate the source code of such iterations with errors. I support your desire to use the latest versions of Qt, but you do not deliver it as a dependency. I am sure that the use of "conan" or "cpm" could solve this problem otherwise be indicated in README.md, that your project can be built exclusively with Qt 5.5.1 source code (at the moment I was able to figure it out). It should be clearly indicated for people trying to compile your product. Otherwise, it should be noted that you have not set up a friendly and were not prepared for the opening of the code.

**Comment by rzhilkin** [comment]
> Unfortunately, the documentation is very sparse and clearly do not understand (i.e. "you need to prepare Qt environment first" really means "you SHOULD compile Qt from source code") in other case require investigate the such errors.

Indeed, if you want to be sure that the project is built without any errors, you should follow readme file. Alternatively, you can ask us for help.

> I support your desire to use the latest versions of Qt, but you do not deliver it as a dependency and it create too many manual operations. (i.e. "conan" or "cpm" may help you to resolve dependency)

We will add more info about Qt dependency in readme file.

> I think that ifdef used in this case may be compromise for backward compatibility, as not all have the latest Qt packages on Linux distros.

I believe it's the only code block, which couldn't be compiled with old Qt5 version. We have fixed it in our repository already and will upload to GitHub soon.

> You basically do not want to support the Qt version of the popular Linux distributions?

Support of Qt versions lower that 5.4 is not planned.

**Comment by vit1251** [comment]
> Indeed, if you want to be sure that the project is built without any errors, you should follow readme file.
> Alternatively, you can ask us for help.

Can I ask you pay attention on "you need to prepare Qt environment first" it does not really means "you SHOULD compile Qt from source code". This phrase is unclear and may confusing.

> We will add more info about Qt dependency in readme file.
Thanx.

> We have fixed it in our repository already and will upload to GitHub soon.

Good. Look forward to that revision.

> Support of Qt versions lower that 5.4 is not planned.

With 5.4 inclusive?

**Comment by rzhilkin** [comment]
> Can I ask you pay attention on "you need to prepare Qt environment first" it does not really means "you SHOULD compile Qt from source code". This phrase is unclear and may confusing. I'm sure modern developer can find hundreds of ways to prepare the Qt environment.

I see. OK, we'll update description.

> With 5.4 inclusive?

5.4 and higher. We compile with Qt 5.4 on Windows, 5.5 on linux and 5.6 on MacOS.

**Comment by vit1251** [comment]
Your README describes steps in a general way when there are different versions of an action.

In your case, all actions are always ambiguous, I think you can use the build system to perform such actions.

For example, I noticed that you were using Python pre-compile resource script, and it is you may like to use Scons or waf to download Qt and compile it.
